### ISSUE TITLE
RubyGems exception with gems.github.com as a source

### ISSUE THREAD

**Issue by sr** [wycats/bundler#18]
<pre>% gem -v       
1.3.3

% ruby -v
ruby 1.8.7 (2009-06-08 patchlevel 173) [i686-linux]

% cat Gemfile
source "http://gems.github.com"
gem "sinatra-sinatra"
gem "thor"

% gem_bundler
Updating source: http://gems.rubyforge.org
Updating source: http://gems.github.com
/usr/lib/ruby/site_ruby/1.8/rubygems/specification.rb:297:in `>': comparison of String with 3 failed (ArgumentError)
        from /usr/lib/ruby/site_ruby/1.8/rubygems/specification.rb:297:in `_load'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/finder.rb:52:in `load'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/finder.rb:52:in `fetch'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/finder.rb:19:in `initialize'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/finder.rb:19:in `each'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/finder.rb:19:in `initialize'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/manifest.rb:56:in `new'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/manifest.rb:56:in `fetch'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/manifest.rb:19:in `install'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/manifest_file.rb:30:in `install'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/cli.rb:18:in `run'
        from /home/simon/.gems/gems/bundler-0.0.1/lib/bundler/cli.rb:6:in `run'
        from /home/simon/.gems/gems/bundler-0.0.1/bin/gem_bundler:4
        from /home/simon/.gems/bin/gem_bundler:19:in `load'
        from /home/simon/.gems/bin/gem_bundler:19</pre>

**Comment by carllerche** [comment]
Interesting, could you try with the latest version of rubygems and see if it still happens. I'll investigate on version 1.3.3

**Comment by carllerche** [comment]
This indeed seems like a problem. We still have to investigate improving source indexes, so we'll investigate this at the same time.

**Comment by technicalpickles** [comment]
I ran into this, and started poking around. It looks like it's something down in RubyGems, or maybe something to do with GitHub's marshalled spec. Take a look at rubygems/specification:296ish (of 1.3.5, but seems to happen with 1.3.3 too):

```
array = Marshal.load str

spec = Gem::Specification.new
spec.instance_variable_set :@specification_version, array[1]

current_version = CURRENT_SPECIFICATION_VERSION
field_count = if spec.specification_version > current_version then
```

array[1] is "2" (the String of 2, that is), whereas current_version is a fixnum, so obviously the comparison would fail.

**Comment by cldwalker** [comment]
This is possibly happening because of a few incorrectly buillt github gemspecs which Rubygem's can't handle when Marshal loading. I've [mentioned it](http://support.github.com/discussions/gems/135-failing-marshal-index-for-httpgemsgithubcom) to github support but haven't heard back yet.

**Comment by carllerche** [comment]
This should be fixed on master & be part of 0.4.0
